src/compiler/checker.ts

```diff
@@ -8965,10 +8966,13 @@ namespace ts {
             if (left.kind === SyntaxKind.SuperKeyword) {
                 return true;
             }
-            // A protected property is accessible in the declaring class and classes derived from it
+            // A protected property is accessible if the property is within the declaring class or classes derived from it
+            const typeClassDeclaration = <ClassLikeDeclaration>getClassLikeDeclarationOfSymbol(type.symbol);
             if (!enclosingClass || !hasBaseType(enclosingClass, declaringClass)) {
-                error(node, Diagnostics.Property_0_is_protected_and_only_accessible_within_class_1_and_its_subclasses, symbolToString(prop), typeToString(declaringClass));
-                return false;
+                if (!isNodeWithinClass(node, typeClassDeclaration)) {

```

Comment:
But in the case of 
```ts
class Base {
    protected x: string;
    method() {
        var b: Base;
    }
}

class Derived1 extends Base {
    method1() {
        var b: Base;
        var d1: Derived1;

        b.x;            // Currently an Error
        d1.x;           // Currently OK
    }
}
```
`b.x` won't be an error anymore, right? Now that the 
`declared class` = `Base`
`Enclosed class` = `Derived1`
`Derived1` has base `Base` - so `b.x` won't be an error anymore.

Or did i misunderstand you?